api/deployments/models/component_status.go

```diff
@@ -31,15 +42,47 @@ func (p ComponentStatus) String() string {
 	return [...]string{"Stopped", "Consistent", "Reconciling", "Restarting", "Outdated"}[p]
 }
 
-func ComponentStatusFromDeployment(deployment *appsv1.Deployment) ComponentStatus {
-	status := ConsistentComponent
+func ComponentStatusFromDeployment(component radixv1.RadixCommonDeployComponent, kd *appsv1.Deployment, rd *radixv1.RadixDeployment) ComponentStatus {
+	if kd == nil || kd.GetName() == "" {
+		return ComponentReconciling
+	}
+	replicasUnavailable := kd.Status.UnavailableReplicas
+	replicasReady := kd.Status.ReadyReplicas
+	replicas := pointers.Val(kd.Spec.Replicas)
+
+	if isCopmonentRestarting(component, rd) {
+		return ComponentRestarting
+	}
+
+	if !owner.VerifyCorrectObjectGeneration(rd, kd, kube.RadixDeploymentObservedGeneration) {
+		return ComponentOutdated
+	}
+
+	if replicas == 0 {
+		return StoppedComponent
+	}
 
-	switch {
-	case deployment.Status.Replicas == 0:
-		status = StoppedComponent
-	case deployment.Status.UnavailableReplicas > 0:
-		status = ComponentReconciling
+	// Check if component is scaling up or down
+	if replicasUnavailable > 0 || replicas < replicasReady {
+		return ComponentReconciling
 	}
 
-	return status
+	return ConsistentComponent
+}
+
+func isCopmonentRestarting(component radixv1.RadixCommonDeployComponent, rd *radixv1.RadixDeployment) bool {
+	restarted := component.GetEnvironmentVariables()[operatordefaults.RadixRestartEnvironmentVariable]
+	if strings.EqualFold(restarted, "") {
```

Comment:
```suggestion
	if restarted == "" {
```
No need for case insesitive comparison